lib/index.js

```diff
@@ -14,110 +15,74 @@ const DEFAULT_OPTIONS = {
     string: '\x1b[32m',
     special: '\x1b[33m',
     bracket: '\x1b[33m',
+    comment: '\x1b[2m\x1b[90m',
     clear: '\x1b[0m'
   }
 }
 
-const SPLIT_CHARS = '[^a-zA-Z_]'
-
 const DEFAULT_KEYWORD = 'default'
 
 const highlighters = [
-  {
-    name: 'keyword',
-    group: 1,
-    regex: new RegExp(`(^|${SPLIT_CHARS})(${keywords.join('|')})(?=${SPLIT_CHARS}|$)`, 'gi')
-  },
-  {
-    name: 'special',
-    regex: /(=|!=|%|\/|\*|-|,|;|:|\+|<|>)/g
-  },
-  {
-    name: 'function',
-    regex: /(\w+?)\(/g,
-    trimEnd: 1
-  },
-  {
-    name: 'number',
-    regex: /(\b\d+(?:\.\d+)?)/g
-  },
-  {
-    name: 'string',
-    regex: /(['](?:\\'|.)*?[']|["](?:\\"|.)*?["]|[`](?:\\`|.)*?[`])/g
-  },
-  {
-    name: 'bracket',
-    regex: /([()])/g
-  }
-]
+  /\b(?<number>  \d+ (?:\.\d+)?  )\b/,
```

Comment:
Are these spaces serving any purpose, or are they just for readability? If the latter I'd rather not have them as I _think_ they'll actually match and I don't think they make it significantly easier to read. In that case we could also remove the `eslint-disable`-comment at the top of the file which would make me happy :smile:

Reply:
The spaces are for readability only (they are removed in `getRegexString()`). Most regex flavors have [a dedicated modifier for this `/x`](https://www.rexegg.com/regex-modifiers.html#freespacing). Unfortunately JS doesn't support that yet, but since we're already processing the regex string, it was trivial to add.

For me spaces makes more complex expressions readable and I can see the structure of the expression. Without the spaces I can't really read it.

Compare:
```
/(?<string>  '(?: [^'\\] | \\. )*'  |  "(?: [^"\\] | \\. )*"  |  `(?: [^`\\] | \\. )*`  )/
```
vs
```
/(?<string>'(?:[^'\\]|\\.)*'|"(?:[^"\\]|\\.)*"|`(?:[^`\\]|\\.)*`)/
```

With spaces I can clearly see that there are three main alternations and what prens balance each other, what's going on inside each non-capturing group, etc.

Without spaces I can't really see anything 😅, and to read it and understand it I would need to add spaces (which is something I do when working with a complex expression).

Do you still want to have them removed?

Reply:
In my opinion, they should be removed.

Reply:
So having thought about it I'd rather have the spaces removed. It also makes it less likely for someone (me) to mess up the formatting in the future since there's not regex linter in place. Thanks!

Reply:
Oh, I didn't realize exactly how the new (and also old) code treats the "default token" as a special case.  I think there are a number of problems with that approach, but I'll bring them up after you merge this PR.

Reply:
@scriptcoded sure, removed the comments and updated the PR.